Sophia.md

```diff
@@ -676,38 +561,30 @@ contract addition =
 
 Sophia can generate blockchain transactions.
 
+### Contracts as objects, calling remotely and in
+Sophia's equivalent of modules is the static contract. Contracts take the place of classes in object-oriented languages, and support local state and will support inheritance (when this feature is implemented).
+
+A contract's public API is the functions and types which are annotated with the `public` keyword. These can be used from outside the contract. All other functions and types are part of the internal API and can be used by contracts inheriting from this contract, except those annotated with `private`, which can only be used locally.
 
-#### Calling a contract function from outside of contract
-Public functions of deployed contracts can be called using `raw_call` function.
-This function is called in following manner-
+#### Calling a function in another contract
+
+The public functions of deployed contracts can be called when you know the address of the contract. The type of the address is a contract. For instance given this abstract contract
+```ocaml
+// An abstract contract
+contract VotingType =
+  public stateful function vote : string => ()
 ```
-raw_call(contract_address: address, function_name: string, gas: int, value: int, args: `a)
+we can call it in the following manner
 ```
-The arguements are:
-- address: Address of the contract. 
-Type = `address`
-- function_name: Name of the function to be called. 
-Type = `string`
-- gas: Gas limit of the transaction. 
-Type = `int`
-- value: ?
-- args: The arguements thefunction takes. If only one, simply pass the value, if more than one, pass them as tuples. 
-Type = if one arguement, can be of any type, if more than one, of type `tuple`.
-
-An example: 
-```javascript
-//call a function "main(arg1)" of contract deployed at contract_address
-// with gas= 10000, value = 10
-contract RemoteCall =
-    function call(contract_address : address, arg1 : int) : int =
-        let gas   = 10000
-        let value = 10
-        raw_call(contract_address, "main", gas, value, arg1)
-
-    function call42(r : address) : int = call(r, 42)
- 
+  function voteTwice(v : VotingType, fee : int, alt : string) =
+    v.vote(value = fee, alt)
+    v.vote(value = fee, alt)
```

Comment:
Maybe say something brief about named arguments, like that you can omit the `value` in which case it defaults to zero.